### ISSUE TITLE
How to define test/validation sets for dynamic datasets

### ISSUE THREAD

**Issue by MaxiBoether** [eth-easl/modyn#49]
It is not clear to me yet how we define test/validation tests in context of dynamic data sets. I am open to discussing this here. In general, we have two cases to consider:

a) Our dataset is dynamic because we just want _more_ data, e.g., in autonomous driving. We do not expect large distribution shifts, but continuously collect new data to improve our model. In this case, a fixed test/validation set will suffice, because we just assume the model to get better.

b) Our dataset is dynamic because we expect distribution shifts. In this case, keeping a static test/validation set is actually counterproductive because if we train on new data, but validate on old data, performance will appear to degrade. Hence, we need to _somehow_ update our validation/test sets. However, this comes with several questions:

- Do we only want to validate on new data? We might want to check how our model performs on old data as well.
- How exactly do we update/create new test and validation sets, especially as we don't know how often and how much data is coming in?

We might want to discuss with our industry partners how they handle this.

In static experiment settings where we replay data, some of these questions might be easier to answer because we can pre-define validation sets. However, even then, we have to make sure that it matches the trigger. For example, if we have a data set over 20 days and define one validation set per day, but trigger training every 1000th new incoming data point, then the trigger (amount of data) will not necessarily be run every day or even multiple times per day. I am not sure how we want to map all of this, conceptually.

W will probably need to have a setting for the pipeline that sets whether the test/validation set should be static or dynamic.

**Comment by MaxiBoether** [comment]
Closing this since we are thinking about this in the context of @XianzheMa's thesis.
